Python/Product/VSCommon/Infrastructure/ComDisposer.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.PythonTools.Infrastructure {
+    internal abstract class ComDisposerBase : IDisposable {
+        protected System.Collections.Generic.List<object> ObjectList { get; set; }
```

Comment:
Full qualification shouldn't be needed here.

Reply:
Also, I think this should be a set, since otherwise an object added twice will also be disposed twice, which is never the right thing to do.